### ISSUE TITLE
support for port 0

### ISSUE THREAD

**Issue by jwilk** [joeyespo/grip#195]
I want to run multiple instances of grip at the same time.
Manually selecting port every time is tedious, so I thought I'll let my OS choose it for me by using port 0.

This works very well through API:
```console
$ python -c 'from grip import serve; serve(port=0)'
 * Running on http://localhost:35739/ (Press CTRL+C to quit)
```
But it doesn't work on the command-line:
```console
$ grip . 0
 * Running on http://0:6419/ (Press CTRL+C to quit)
```
(Um, no, that's not what I wanted! :unamused:)

```console
$ grip . localhost:0
Error: Invalid address 'localhost:0'
 * Running on http://localhost:6419/ (Press CTRL+C to quit)
```
(Still not what I wanted…)

**Comment by joeyespo** [comment]
Didn't know about this! Very cool. Fixed in #196.

**Comment by joeyespo** [comment]
Thanks for the issue, @jwilk!

**Comment by joeyespo** [comment]
FYI this was released with v4.3.1.
